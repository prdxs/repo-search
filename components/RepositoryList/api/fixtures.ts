import { IRepository } from './typings';

export const repositories: Array<IRepository> = [
  {
    id: '0',
    name: 'facebook/react',
    link: 'https://github.com/facebook/react',
    stars: 151026,
    forks: 29654,
    issues: 498,
    lastUpdated: '1st Jul 2020 07:20:59',
  },
  {
    id: '1',
    name: 'prdxs/repo-search',
    link: 'https://github.com/prdxs/repo-search',
    stars: 0,
    forks: 0,
    issues: 0,
    lastUpdated: '1st Jul 2020 14:20:17',
  },
];
