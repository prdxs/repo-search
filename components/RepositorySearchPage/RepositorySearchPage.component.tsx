import React from 'react';
import Link from 'next/link';
import clsx from 'clsx';
import styled, { css } from 'styled-components';
import { transparentize } from 'polished';
import Grid from '@material-ui/core/Grid';
import CircularProgress from '@material-ui/core/CircularProgress';
import Fab from '@material-ui/core/Fab';
import VisibilityIcon from '@material-ui/icons/Visibility';

import { IRepositorySearchPageProps } from './RepositorySearchPage.typings';
import SearchBar from '@/components/SearchBar';
import RepositoryCard from '@/components/RepositoryCard';
import getRepoFromUrl from '@/utils/getRepoFromUrl';

const RepositorySearchPage: React.FC<IRepositorySearchPageProps> = ({
  className,
  style,
  loading,
  repositories,
  query,
  onChange,
  onClear,
  withLinks,
}) => (
  <div className={clsx('RepositorySearchPage-root', className)} style={style}>
    <Grid container justify="center">
      <Grid item>
        <SearchBar value={query} onChange={onChange} onClear={onClear} />
      </Grid>
    </Grid>
    <Grid container justify="center">
      <Grid item className="RepositorySearchPage-list">
        <Grid container direction="column" spacing={4}>
          {loading && (
            <Grid item>
              <Grid container justify="center">
                <Grid item>
                  <CircularProgress />
                </Grid>
              </Grid>
            </Grid>
          )}
          {repositories.length > 0
            ? repositories.map((repo) => {
                const viewButton = (
                  <Fab
                    className="RepositorySearchPage-viewButton"
                    color="primary"
                    aria-label="view"
                    size="small"
                  >
                    <VisibilityIcon />
                  </Fab>
                );
                const { owner, repo: repoName } = getRepoFromUrl(repo.link);

                return (
                  <Grid item key={repo.id}>
                    <div className="RepositorySearchPage-cardContainer">
                      <RepositoryCard
                        name={repo.name}
                        link={repo.link}
                        stars={repo.stars}
                        forks={repo.forks}
                        issues={repo.issues}
                        lastUpdated={repo.lastUpdated}
                      />
                      {withLinks ? (
                        <Link
                          href="/repositories/[id]/[owner]/[repo]"
                          as={`/repositories/${repo.id}/${owner}/${repoName}`}
                        >
                          {viewButton}
                        </Link>
                      ) : (
                        viewButton
                      )}
                    </div>
                  </Grid>
                );
              })
            : null}
        </Grid>
      </Grid>
    </Grid>
  </div>
);

RepositorySearchPage.defaultProps = {
  withLinks: false,
};

const StyledRepositorySearchPage = styled(RepositorySearchPage)`
  &.RepositorySearchPage-root {
    padding-top: calc(${({ theme }) => theme.spacing(24)} + 52px);
    height: 100vh;

    > .MuiGrid-container:nth-child(1) {
      position: fixed;
      top: 0;
      left: 0;
      z-index: 100;
      width: 100%;
      padding-top: ${({ theme }) => theme.spacing(8)};
      padding-bottom: ${({ theme }) => theme.spacing(16)};

      ${({ theme }) => {
        const color = theme.palette.background.default;
        const fromColor = transparentize(1.0, color);
        const toColor = transparentize(0.0, color);

        return css`
          background-image: linear-gradient(to top, ${fromColor}, ${toColor});
        `;
      }}
    }

    .RepositorySearchPage-list {
      max-width: 500px;
      padding: ${({ theme }) => theme.spacing(0, 2, 8)};

      .RepositorySearchPage-cardContainer {
        position: relative;

        .RepositorySearchPage-viewButton {
          position: absolute;
          right: -48px;
          bottom: 0;
          opacity: 0;
          transition: opacity 0.4s;
        }

        &:hover .RepositorySearchPage-viewButton {
          opacity: 1;
        }
      }
    }
  }
`;

export default StyledRepositorySearchPage;
