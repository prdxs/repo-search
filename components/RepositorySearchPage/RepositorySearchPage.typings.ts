import { IComponentProps } from '@/typings/common';
import { IRepository } from './state/typings';

export interface IRepositorySearchPageProps extends IComponentProps {
  repositories: Array<IRepository>;
  loading: boolean;
  query: string;
  onChange(
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ): void;
  onClear(): void;
}
