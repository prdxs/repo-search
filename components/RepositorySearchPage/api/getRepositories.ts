import { SearchReposResponseData } from '@octokit/types';
import moment from 'moment';

import { ArrayElement } from '@/typings/utils';
import octokit from '@/utils/octokit';

import { IRepository } from '@/components/RepositorySearchPage/state/typings';

const mapDataItemsToRepos = (
  item: ArrayElement<SearchReposResponseData['items']>
): IRepository => ({
  id: item.node_id,
  name: item.name,
  stars: item.stargazers_count,
  link: item.html_url,
  forks: item.forks_count,
  issues: item.open_issues_count,
  lastUpdated: moment(item.updated_at).format('Do MMM YYYY HH:mm:ss'),
});

const getRepositories = async (q: string): Promise<Array<IRepository>> => {
  const { data: dataByUser } = await octokit.search.repos({ q: `user:${q}` });
  const { data: dataByKeyword } = await octokit.search.repos({ q });

  const byUserIds = dataByUser.items.map((item) => item.node_id);
  const reposByUser = dataByUser.items.map(mapDataItemsToRepos);
  const reposByKeyword = dataByKeyword.items
    .filter((item) => byUserIds.includes(item.node_id))
    .map(mapDataItemsToRepos);

  return [...reposByUser, ...reposByKeyword];
};

export default getRepositories;
