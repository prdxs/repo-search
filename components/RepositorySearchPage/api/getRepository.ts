import { ReposGetResponseData } from '@octokit/types';
import moment from 'moment';

import octokit from '@/utils/octokit';

import { IRepository } from '@/components/RepositorySearchPage/state/typings';

const mapDataItemToRepo = (item: ReposGetResponseData): IRepository => ({
  id: item.node_id,
  name: item.name,
  stars: item.stargazers_count,
  link: item.html_url,
  forks: item.forks_count,
  issues: item.open_issues_count,
  lastUpdated: moment(item.updated_at).format('Do MMM YYYY HH:mm:ss'),
});

const getRepository = async (
  owner: string,
  repo: string
): Promise<IRepository> => {
  const { data } = await octokit.repos.get({ owner, repo });
  return mapDataItemToRepo(data);
};

export default getRepository;
