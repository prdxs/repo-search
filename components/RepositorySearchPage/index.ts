export { default } from './RepositorySearchPage.container';
