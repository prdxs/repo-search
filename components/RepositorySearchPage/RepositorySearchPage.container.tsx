import React, { useCallback, useEffect, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import debounce from 'debounce';

import { IComponentProps } from '@/typings/common';
import RepositorySearchPage from './RepositorySearchPage.component';
import { searchRepositories } from './state/actions';
import {
  selectRepositories,
  selectRepositoriesLoading,
} from './state/selectors';

const RepositorySearchPageContainer: React.FC<
  IComponentProps & { withLinks: boolean }
> = ({ withLinks, ...otherProps }) => {
  const dispatch = useDispatch();
  const loading = useSelector(selectRepositoriesLoading);
  const repositories = useSelector(selectRepositories);

  const [query, setQuery] = useState('');

  const handleChange = useCallback(
    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
      setQuery(event.target.value);
    },
    []
  );

  const handleClear = useCallback(() => {
    setQuery('');
  }, []);

  useEffect(() => {
    const debouncedSearch = debounce(() => {
      dispatch(searchRepositories(query));
    }, 500);

    debouncedSearch();

    return () => {
      debouncedSearch.clear();
    };
  }, [query]);

  return (
    <RepositorySearchPage
      query={query}
      loading={loading}
      repositories={repositories}
      onChange={handleChange}
      onClear={handleClear}
      withLinks={withLinks}
      {...otherProps}
    />
  );
};

export default RepositorySearchPageContainer;
