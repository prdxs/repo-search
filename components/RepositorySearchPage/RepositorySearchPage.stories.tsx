import React from 'react';

import StorybookWrapper from '@/components/StorybookWrapper';
import RepositorySearchPage from './RepositorySearchPage.component';
import { repositories } from './api/fixtures';

export default {
  title: 'RepositorySearchPage',
  component: RepositorySearchPage,
};

export const Default: React.FC = () => {
  return (
    <StorybookWrapper>
      <RepositorySearchPage repositories={repositories} />
    </StorybookWrapper>
  );
};
