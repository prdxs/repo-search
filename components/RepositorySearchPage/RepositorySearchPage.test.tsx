import React from 'react';
import { render, fireEvent } from '@/test/utils';

import RepositorySearchPage from './RepositorySearchPage.component';
import { repositories } from './api/fixtures';

describe('RepositorySearchPage', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(
      <RepositorySearchPage
        loading={true}
        repositories={repositories}
        query=""
        onChange={jest.fn()}
        onClear={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render as many repositories as provided', () => {
    const { getAllByTestId } = render(
      <RepositorySearchPage
        loading={false}
        repositories={repositories}
        query=""
        onChange={jest.fn()}
        onClear={jest.fn()}
      />
    );

    const repositoryEls = getAllByTestId('RepositoryCard-root');

    expect(repositoryEls).toHaveLength(2);
  });

  it('should render a SearchBar with the value provided in the query prop', () => {
    const query = 'prdxs';
    const { getByDisplayValue } = render(
      <RepositorySearchPage
        loading={false}
        repositories={repositories}
        query={query}
        onChange={jest.fn()}
        onClear={jest.fn()}
      />
    );

    const inputEl = getByDisplayValue(query);

    expect(inputEl).toBeInTheDocument();
  });

  it('should call change handler when the value changes', () => {
    const query = 'prdxs';
    const handleChange = jest.fn();
    const { getByTestId } = render(
      <RepositorySearchPage
        loading={false}
        repositories={repositories}
        query=""
        onChange={handleChange}
        onClear={jest.fn()}
      />
    );
    const input = getByTestId('SearchBar-actualInput');

    fireEvent.change(input, { target: { value: query } });

    expect(handleChange).toHaveBeenCalled();
  });

  it('should call clear handler when the clear button is clicked', () => {
    const handleClear = jest.fn();
    const { getByTestId } = render(
      <RepositorySearchPage
        loading={false}
        repositories={repositories}
        query="prdxs"
        onChange={jest.fn()}
        onClear={handleClear}
      />
    );
    const clearButtonEl = getByTestId('SearchBar-clearButton');

    fireEvent.click(clearButtonEl);

    expect(handleClear).toHaveBeenCalled();
  });
});
