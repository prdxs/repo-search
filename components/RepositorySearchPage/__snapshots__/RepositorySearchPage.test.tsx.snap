// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositorySearchPage should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    style="padding: 2rem;"
  >
    <div
      class="RepositorySearchPage-root RepositorySearchPagecomponent__StyledRepositorySearchPage-sc-1vhjeex-0 fboBxR"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <div
            class="MuiPaper-root SearchBar-root SearchBarcomponent__StyledSearchBar-b8c7be-0 ixSzUK MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiInputBase-root SearchBar-input"
              data-testid="SearchBar-input"
            >
              <input
                aria-label="search github users"
                class="MuiInputBase-input"
                data-testid="SearchBar-actualInput"
                placeholder="Search Github users"
                type="text"
                value=""
              />
            </div>
            <button
              aria-label="search"
              class="MuiButtonBase-root MuiIconButton-root .SearchBar-iconButton"
              data-testid="SearchBar-searchButton"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
      >
        <div
          class="MuiGrid-root RepositorySearchPage-list MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-direction-xs-column"
          >
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
              >
                <div
                  class="MuiGrid-root MuiGrid-item"
                >
                  <div
                    class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                    role="progressbar"
                    style="width: 40px; height: 40px;"
                  >
                    <svg
                      class="MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                        cx="44"
                        cy="44"
                        fill="none"
                        r="20.2"
                        stroke-width="3.6"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="RepositorySearchPage-cardContainer"
              >
                <div
                  class="MuiPaper-root MuiCard-root RepositoryCard-root RepositoryCardcomponent__StyledRepositoryCard-sc-1c2lx1b-0 ljLEiR MuiPaper-elevation1 MuiPaper-rounded"
                  data-testid="RepositoryCard-root"
                >
                  <div
                    class="MuiCardHeader-root"
                  >
                    <div
                      class="MuiCardHeader-content"
                    >
                      <span
                        class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4z"
                              />
                            </svg>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6"
                            >
                              <a
                                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
                                href="https://github.com/facebook/react"
                                target="_blank"
                              >
                                facebook/react
                              </a>
                            </h6>
                          </div>
                        </div>
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiCardContent-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root RepositoryCard-star MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorPrimary"
                              >
                                151026
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                              >
                                498
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M14 4l2.29 2.29-2.88 2.88 1.42 1.42 2.88-2.88L20 10V4zm-4 0H4v6l2.29-2.29 4.71 4.7V20h2v-8.41l-5.29-5.3z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                              >
                                29654
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <span
                      class="MuiTypography-root RepositoryCard-lastUpdated MuiTypography-caption"
                    >
                      1st Jul 2020 07:20:59
                    </span>
                  </div>
                </div>
                <button
                  aria-label="view"
                  class="MuiButtonBase-root MuiFab-root RepositorySearchPage-viewButton MuiFab-sizeSmall MuiFab-primary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item"
            >
              <div
                class="RepositorySearchPage-cardContainer"
              >
                <div
                  class="MuiPaper-root MuiCard-root RepositoryCard-root RepositoryCardcomponent__StyledRepositoryCard-sc-1c2lx1b-0 ljLEiR MuiPaper-elevation1 MuiPaper-rounded"
                  data-testid="RepositoryCard-root"
                >
                  <div
                    class="MuiCardHeader-root"
                  >
                    <div
                      class="MuiCardHeader-content"
                    >
                      <span
                        class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <svg
                              aria-hidden="true"
                              class="MuiSvgIcon-root"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M18 2H6c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zM6 4h5v8l-2.5-1.5L6 12V4z"
                              />
                            </svg>
                          </div>
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <h6
                              class="MuiTypography-root MuiTypography-h6"
                            >
                              <a
                                class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorSecondary"
                                href="https://github.com/prdxs/repo-search"
                                target="_blank"
                              >
                                prdxs/repo-search
                              </a>
                            </h6>
                          </div>
                        </div>
                      </span>
                    </div>
                  </div>
                  <div
                    class="MuiCardContent-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-container"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root RepositoryCard-star MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorPrimary"
                              >
                                0
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M20 8h-2.81c-.45-.78-1.07-1.45-1.82-1.96L17 4.41 15.59 3l-2.17 2.17C12.96 5.06 12.49 5 12 5c-.49 0-.96.06-1.41.17L8.41 3 7 4.41l1.62 1.63C7.88 6.55 7.26 7.22 6.81 8H4v2h2.09c-.05.33-.09.66-.09 1v1H4v2h2v1c0 .34.04.67.09 1H4v2h2.81c1.04 1.79 2.97 3 5.19 3s4.15-1.21 5.19-3H20v-2h-2.09c.05-.33.09-.66.09-1v-1h2v-2h-2v-1c0-.34-.04-.67-.09-1H20V8zm-6 8h-4v-2h4v2zm0-4h-4v-2h4v2z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                              >
                                0
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-item"
                          >
                            <span
                              class="MuiBadge-root"
                            >
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M14 4l2.29 2.29-2.88 2.88 1.42 1.42 2.88-2.88L20 10V4zm-4 0H4v6l2.29-2.29 4.71 4.7V20h2v-8.41l-5.29-5.3z"
                                />
                              </svg>
                              <span
                                class="MuiBadge-badge MuiBadge-anchorOriginBottomRightRectangle MuiBadge-colorSecondary"
                              >
                                0
                              </span>
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                    <span
                      class="MuiTypography-root RepositoryCard-lastUpdated MuiTypography-caption"
                    >
                      1st Jul 2020 14:20:17
                    </span>
                  </div>
                </div>
                <button
                  aria-label="view"
                  class="MuiButtonBase-root MuiFab-root RepositorySearchPage-viewButton MuiFab-sizeSmall MuiFab-primary"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiFab-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
